app/models/foreman_openscap/arf_report.rb

```diff
@@ -130,10 +136,12 @@ def othered?
     end
 
     def equal?(other)
-      results = [xccdf_rule_results, other.xccdf_rule_results].flatten.group_by(&:xccdf_rule_id).values
+      results = [logs, other.logs].flatten.group_by(&:source_id).values
       # for each rule, there should be one result from both reports
       return false unless results.map(&:length).all? { |item| item == 2 }
-      results.map { |result| result.first.xccdf_result_id == result.last.xccdf_result_id }.all? && asset_id == other.asset_id && policy_id == other.policy_id
+      results.map { |result| result.first.source_id == result.last.source_id }.all? &&
```

Comment:
I know it's not your code originally, but the first condition part should be equivalent to
```ruby
results.all? { |result| result.first.source_id == result.last.source_id } &&
```

but more effective, since it stops on first mismatch